jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1477,6 +1477,27 @@ private char compareValue(byte b) {
 		return Character.toLowerCase(Character.toUpperCase(helpers.byteToCharUnsigned(b)));
 	}
 
+	private boolean charValuesEqual(char c1, char c2) {
+		boolean charValuesEqual = false;
+
+		if (c1 == c2) {
+			charValuesEqual = true;
+		} else {
+			// If at least one char is ASCII, converting to upper cases then compare should be sufficient.
+			// If both chars are not in ASCII char set, need to convert to lower case and compare as well.
+			char c1upper = (char) toUpperCase(c1);
+			char c2upper = (char) toUpperCase(c2);
+
+			if (((c1 <= 255 || c2 <= 255) && (c1upper == c2upper))
+					|| (toLowerCase(c1upper) == toLowerCase(c2upper))
+			) {
+				charValuesEqual = true;
+			}
+		}
+
+		return charValuesEqual;
+	}
+
```

Comment:
@pshipton Here's the new helper. This is now being used anytime we are checking equality for two chars in `equalsIgnoreCase()` and `regionMatches()`.

Reply:
Local testing is fine. I'll be running PR testing anyway.